### PR TITLE
KAFKA-5541: minor follow-up

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @tedyu @guozhangwang @dguy @bbejeck

**Comment by tedyu** [comment]
lgtm

**Review by bbejeck — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java [hunk 1]
```diff
@@ -512,7 +512,7 @@ void close(final boolean clean) {
                           task.getClass().getSimpleName(),
                           task.id(),
                           t);
-                firstException.compareAndSet(null, closeUncleanIfRequired(task, clean));
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java:515]: I thought we planned to no throw the exception if the unclean closure succeeds, in which case we should not remember the exception in line 510 but only here 515?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java:515]: Sounds correct.

**Comment by mjsax** [comment]
Updated and rebased.

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk.

**Pull request closed without merging**